### PR TITLE
[skip e2e]Update mergify rule for 2.*.* branch

### PR DESCRIPTION
**Pull request by jennyli-z**
Signed-off-by: Jenny Li <jing.li@zilliz.com>
/kind improvement

### CODE REVIEW

File: .github/mergify.yml [hunk 1]
```diff
@@ -1,7 +1,9 @@
 pull_request_rules:
   - name: Add needs-dco label when DCO check failed
     conditions:
-      - base=master
+      - or:
+        - base=master
+        - base~=2+(.\d+){2}$
```
Comment (congqixia) [inline .github/mergify.yml:6]: Maybe should be 
```
(\.\d+)
```

Comment (jennyli-z) [reply .github/mergify.yml:6]: Any reason？ I have tested on my repo & it works for 2.0.2.

**Comment by jennyli-z** [comment]
@congqixia updated. please help review again.

**Comment by congqixia** [comment]
lgtm, please verify @LoveEachDay 
/approve

**Review by LoveEachDay — APPROVED** [review]
/lgtm

**Pull request merged**